src/nvim/channel.c

```diff
@@ -639,7 +639,11 @@ static void channel_process_exit_cb(Process *proc, int status, void *data)
     rpc_close(chan);
   }
 
-  process_job_event(chan, &chan->on_exit, "exit", NULL, 0, status);
+  // if status is -1 the process did not really exit,
+  // we just closed the handle onto a detached process
```

Comment:
What if the process return -1?

Reply:
Doesn't POSIX only guarantee 0-255 works? If I try `int main() { return -1; }` I get 255 back.